vizro-mcp/src/vizro_mcp/_schemas/schemas.py

```
@@ -137,7 +63,82 @@ class AgGridEnhanced(vm.AgGrid):
+    try:
+        function_name = value["_target_"]
+    except KeyError as exc:
+        raise ValueError(
+            "CapturedCallable object must contain the key '_target_' that gives the target function."
+        ) from exc
+
+    try:
+        _ = getattr(importlib.import_module(import_path), function_name)
+    except (AttributeError, ModuleNotFoundError):
+        pass
+    else:
+        if function_name in context.get("custom_chart_defs", {}):
+            raise ValueError(
+                f"""_target_={function_name} is defined as a `custom_chart` and can be imported from {import_path}.
+                Do not use names for custom charts that are already in the px namespace.
+                """
+            )
+        return CapturedCallable._validate_captured_callable(
+            captured_callable_config=value, json_schema_extra=json_schema_extra
+        )
+
+    if function_name in context.get("custom_chart_defs", {}):
+        items_str = ", ".join(f"{k}={v!r}" for k, v in value.items() if k != "_target_")  # type: ignore
+        return f"{value['_target_']}({items_str})"
+
+    raise ValueError(
+        f"""_target_={function_name} cannot be imported from {import_path} nor is it defined as a `custom_chart`
+        with the correct name."""
+    )
+
+
+class Graph(vm.Graph):
+    figure: Annotated[
+        Union[CapturedCallable, str],
+        # AfterValidator(process_callable_data_frame), #pretty sure this is not needed for this task
+        Field(
+            json_schema_extra={"mode": "graph", "import_path": "vizro.plotly.express"},
+            description=""""Function that returns a plotly `go.Figure`""",
+        ),
+    ]
+    _validate_figure = field_validator("figure", mode="before")(validate_captured_callable)
+
+
+####### This needs improvement #######
+# TODO: can this be done cleaner? This currently fails for Graphs in Containers in Tabs!!
```

in this very special edge case, could the validator catch this edge case and ignore the error?

---

The problem lies deeper I think - in this case the new validator is not used, but the validator of the original model. While we can ignore ValidationErrors, we then would not have an instantiated model...